src/hsl.rs

```diff
@@ -13,70 +16,70 @@ use {Color, Rgb, Luma, Xyz, Lab, Lch, Hsv, ColorSpace, Mix, Shade, GetHue, Hue,
 ///
 ///See [HSV](struct.Hsv.html) for a very similar color space, with brightness instead of lightness.
 #[derive(Clone, Copy, Debug, PartialEq)]
-pub struct Hsl {
+pub struct Hsl<T: Float = f32> {
     ///The hue of the color, in degrees. Decides if it's red, blue, purple,
     ///etc.
-	pub hue: RgbHue,
+    pub hue: RgbHue<T>,
 
-    ///The colorfulness of the color. 0.0 gives gray scale colors and 1.0 will
+    ///The colorfulness of the color. T::zero() gives gray scale colors and T::one() will
     ///give absolutely clear colors.
-	pub saturation: f32,
+    pub saturation: T,
 
-    ///Decides how light the color will look. 0.0 will be black, 0.5 will give
-    ///a clear color, and 1.0 will give white.
-	pub lightness: f32,
+    ///Decides how light the color will look. T::zero() will be black, T::from(0.5).unwrap() will give
```

Comment:
I wonder if it would be better to keep these as they were. I don't think people will be more confused by `0.0` and `1.0` than `T::zero()` and `T::one()` ...and `T::from(0.5).unwrap()` looks very awkward to have in the documentation, to be honest.